src/controllers/expenses.js

```diff
@@ -0,0 +1,170 @@
+'use strict';
+
+const expenseService = require('../services/expenses');
+const userService = require('../services/users');
+const { isDateValid } = require('../functions/validateDate');
+
+const getAll = (req, res, _next) => {
+  if (Object.keys(req.query).length === 0) {
+    const expenses = expenseService.getAll();
+
+    res.send(expenses);
+
+    return;
+  }
+
+  const {
+    userId,
+    categories,
+    from,
+    to,
+  } = req.query;
+
+  if (userId && !categories && !from && !to) {
+    const foundUser = userService.getById(userId);
+
+    if (!foundUser) {
+      res.sendStatus(404);
+
+      return;
+    }
+  }
+
+  let categoriesArray = categories;
+
+  if (categoriesArray) {
+    categoriesArray = Array.isArray(categories)
+      ? categories
+      : [ categories ];
+  }
+
+  const foundExpenses = expenseService
+    .getByParameters(userId, categoriesArray, from, to);
+
+  res.send(foundExpenses);
+};
+
+const getOne = (req, res) => {
+  const { expenseId } = req.params;
+  const foundExpense = expenseService.getById(expenseId);
+
+  if (!foundExpense) {
+    res.sendStatus(404);
+  } else {
+    res.send(foundExpense);
+  }
+};
+
+const add = (req, res) => {
+  const {
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  } = req.body;
+
+  const foundUser = userService.getById(userId);
+
+  if (!foundUser) {
+    res.sendStatus(400);
+
+    return;
+  };
+
+  if (
+    typeof userId !== 'number'
+    || !isDateValid(spentAt)
+    || typeof title !== 'string'
+    || typeof amount !== 'number'
+    || typeof category !== 'string'
+    || typeof note !== 'string'
+  ) {
+    res.sendStatus(400);
+
+    return;
+  }
```

Comment:
Nice work validating requests.
The issues is, request parameters should be checked before you query your database (or in-memory model). Move this check before `userService.getById(userId);`.

Also this applies to all of your endpoints, make sure to move validation before your queries/searches/deletes/etc.